node/src/chain.rs

```diff
@@ -130,11 +130,13 @@ pub fn create_substreams_networks(
                     .entry(chain.protocol)
                     .or_insert_with(FirehoseNetworks::new);
 
-                for i in 0..firehose.conn_pool_size {
+                for _ in 0..firehose.conn_pool_size {
                     parsed_networks.insert(
                         name.to_string(),
                         Arc::new(FirehoseEndpoint::new(
-                            &format!("{}-{}", provider.label, i),
+                            // This label needs to be the original label so that the metrics
+                            // can be deduped.
+                            &provider.label,
```

Comment:
This is a functional change right, instead of segregating the metrics by individual conns, they are grouped for the whole pool? Makes sense to me, but just confirming.

Reply:
The metrics will be tagged with provider/host, result, request type and so on. I changed from hosts to providers which means that the dedup needs to happen on the provider level (eg, if one conn of providerA fails, all of the providerA conns should be de-prioritised until checked again).